lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -160,6 +160,10 @@ defmodule Mix.Tasks.Compile.App do
         unless is_list(value) do
           Mix.raise "Application description (:description) is not a character list, got: #{inspect value}"
         end
+
+        unless Enum.all?(value, fn char -> char in 0..127 end) do
+          Mix.raise "Application description (:description) may contain only ASCII-printable codepoints, got: #{inspect value}"
```

Comment:
I think we should not say `printable` since we check for every ASCII code and they all are allowed.